clients/privacy-center/components/consent/notice-driven/NoticeDrivenConsent.tsx

```diff
@@ -307,53 +302,139 @@ const NoticeDrivenConsent = ({ base64Cookie }: { base64Cookie: boolean }) => {
-  return (
-    <Stack spacing={6} paddingX={12}>
-      {items.map((item, index) => {
-        const { id, highlight, url, name, description, historyId, disabled } =
-          item;
-
-        const handleChange = (value: boolean) => {
-          const pref = value
-            ? UserConsentPreference.OPT_IN
-            : UserConsentPreference.OPT_OUT;
-          setDraftPreferences({
-            ...draftPreferences,
-            ...{ [historyId]: pref },
-          });
-        };
+  const handleConsentToggleChange = (value: boolean, historyId: string) => {
+    const pref = value
+      ? UserConsentPreference.OPT_IN
+      : UserConsentPreference.OPT_OUT;
 
-        return (
-          <React.Fragment key={id}>
-            {index > 0 ? <Divider /> : null}
-            <ConsentItem
+    setDraftPreferences((draftPreferencesState) => ({
+      ...draftPreferencesState,
+      ...{ [historyId]: pref },
+    }));
+  };
+
+  return (
+    <Box width={{ base: "full", lg: "700px" }}>
+      <Accordion allowMultiple mb={4}>
+        {items.map((item) => {
+          const { id, name, description, historyId, disabled } = item;
+
+          return (
+            <ConsentItemAccordion
+              key={id}
               id={id}
-              name={item.bestTranslation?.title || name}
+              title={item.bestTranslation?.title || name}
               description={item.bestTranslation?.description || description}
-              highlight={highlight}
-              url={url}
               value={item.value}
               gpcStatus={item.gpcStatus}
-              onChange={handleChange}
+              onChange={(value) => {
+                handleConsentToggleChange(value, historyId);
+
+                // if has children, toggle all children the same as parent
+                if (item.children && item.children.length > 0) {
+                  item.children.forEach((child) => {
+                    handleConsentToggleChange(value, child.historyId);
+                  });
+                }
+              }}
               disabled={disabled}
-            />
-          </React.Fragment>
-        );
-      })}
+            >
+              <Stack
+                mt={4}
+                borderBottom={1}
+                divider={<StackDivider color="gray.300" />}
+              >
+                {item.children &&
+                  item.children.map((child) => {
+                    return (
+                      <ConsentChildItem
+                        key={child.id}
+                        title={child.name}
+                        id={child.id}
+                        value={child.value}
+                        onChange={(childValue) => {
+                          // save child item consent
+                          handleConsentToggleChange(
+                            childValue,
+                            child.historyId,
+                          );
+
+                          // if child is toggle on, toggle parent on
```

Comment:
This detailed toggle functionality shows true thought and polish in the app. Nice work!